wp-content/civi-extensions/goonjcustom/Civi/DroppingCenterService.php

```diff
@@ -70,6 +141,90 @@ private static function generateDroppingCenterQrCode($id) {
     self::generateQrCode($data, $id, $saveOptions);
   }
 
+  /**
+   *
+   */
+  public static function setOfficeDetails($op, $groupID, $entityID, &$params) {
+    if ($op !== 'create' || self::getSubtypeNameByEntityId($entityID) !== self::ENTITY_SUBTYPE_NAME) {
```

Comment:
I created a new trait to retrieve the subtype name because the setOfficeDetails function doesn't provide the subtype ID. However, I do get the entity ID, so I’m using that to fetch the subtype name. The setOfficeDetails function should work for both the Dropping Center and Collection Camp.